test/bin/compiler-beta.t/run.t

```diff
@@ -0,0 +1,132 @@
+We have a simple project that only depends on OCaml
```

Comment:
Since the code doesn't do anything compiler-specific (which is a good thing!) maybe it would make sense to just set up dummy packages? Dummy packages can have more monimal `opam` files that are easier to grasp and review.

Reply:
That would work but I like the fact that it shows it solves a real world problem, with real world metadata.

Reply:
I agree in principle, but in practice I find these files very long and distracting to read (if there is something wrong, do you want to go through the whole file and try to understand what each of these stanzas does and how it interacts with the others?), so to review it I would need to read through every stanza, understand it, see if it is important etc. I would prefer the tests to concentrate on one aspect and show it working.

Kinda like we want to use minimal examples for tests, so reading and modifying these is easy instead of dealing with accidental complexity.

Reply:
I think here it's important that it also checks that it works with the `ocaml` and `ocaml-base-compiler` split. If we change this test we can still break this behaviour while having the dummy tests pass. I'm happy to extract another test just for the avoid-version if you'd like but this test will stay as a regression test.